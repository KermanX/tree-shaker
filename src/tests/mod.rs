use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
