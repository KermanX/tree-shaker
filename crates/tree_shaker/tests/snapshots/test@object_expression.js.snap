---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/object_expression.js
---
export function main() {
	effect(1, 3);
	let __unused_CE6D = effect();
	let obj3 = { get a() {
		effect();
	} };
	effect(2);
	obj3.a;
	effect(void 0, 3);
	let __unused_6286 = { ...{ get a() {
		effect();
	} } };
	effect(1);
}
export function modify_self_in_getter() {
	const { a: __unused_3616, b: changed } = { get a() {
		this.b = {};
	} };
	return changed;
}
